src/componmnet/PointCloud/PointCloud.jsx

```diff
@@ -1,55 +1,47 @@
 import React, { Component, useEffect, useRef, useState } from 'react';
-import { main, updateLayout } from './render';
-import oriData from '../../../data/data.json';
-import { preprocess } from '../../utils';
-import { Graph } from 'igraph-wasm';
+import { main, updateCamera } from './render';
+import oriData from '../../../data/pca_3dumap_outputs_with_metadata.json';
 
 
-function PointCloud({ layout }) {
-    const [graph, setGraph] = useState(null);
+function PointCloud({ node }) {
+    const canvasRef = useRef(null);
+    const camera = useRef(null);
+    const controls = useRef(null);
+    const nodePool = getNodePool();
 
-    useEffect(async () => {
-        const g = new Graph();
-        await g.init();
-        const graph = g.createInstance();
-        setGraph(graph);
-    }, []);
 
-    const canvasRef = useRef(null);
-    const shperes = useRef(null);
-    useEffect(async () => {
-        if (canvasRef.current !== null && graph) {
-            let data = preprocess(oriData);
-            const spheres = main(canvasRef.current, data, graph);
-            shperes.current = spheres;
+    function getNodePool() {
```

Comment:
`getNodePool` should take `oriData` as an input.
```suggestion
    function getNodePool(oriData) {
```